keypair/keypair.go

```diff
@@ -0,0 +1,84 @@
+package keypair
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"io/ioutil"
+
+	"github.com/mitchellh/go-homedir"
+	"github.com/pkg/errors"
+)
+
+func ParsePrivateKey(privateKeyPath string) (*rsa.PrivateKey, error) {
```

Comment:
I wonder if it is worth making this an RSA package rather than keypair -- since all the functions are specific to RSA.

The alternative is probably adding RSA to the name of these functions -- for example ParsePrivateKey -> ParseRSAPrivateKey